_core/smart/tutorial.md

```diff
@@ -1,308 +1,208 @@
+1. Still inside the terminal, list the contents of the `contracts` folder.
+
+    ```bash
+    ls contracts
     ```
 
-    This happens because the `names.clar` contract _calls_ the `tokens.clar` contract, and that contract has not been created on the blockchain.
+    This directory contains one file for the hello world smart contract. Note that all Clarity files have a `.clar` suffix.
 
-3. Type check the `tokens.clar` contract, it should pass a check as it does not use the `contract-call` function:
+2. Let's review the contents of `hello-world.clar` with the `cat` command.
 
     ```bash
-    # clarity-cli check sample-programs/tokens.clar /data/db
-    Checks passed.
+    cat contracts/hello-world.clar
     ```
 
-    When the `check` command executes successfully and exits with the stand UNIX `0` exit code.
-
-4. Generate a demo Stacks address for testing your contract.
+    You should see the contract source code. Take a few seconds to review the content.
 
-   This address is used to name your contract at launch time. You can use any existing Stacks address. For this sample, you are going to use the `generate_address` command to create one.
+    Clarity is a programming language based on [LISP](https://en.wikipedia.org/wiki/Lisp_(programming_language)). Most notably, Clarity is designed for static analysis, not compiled, and **not** [Turing complete](https://en.wikipedia.org/wiki/Turing_completeness).
 
-   ```bash
-   # clarity-cli generate_address
-   SP28Z69HE5H70BVRG4VGKN4SYNVJ1J0417WVCKZWM
-   ```
+    Let's go through the source code. Notice how the program and each statement is enclosed in `()` (parentheses). You'll see that the smart contract consists of two public functions. Starting at the top, let's review line by line:
 
-   The demo address you generate will be different than the one that appears in this example. 
+    ```cl
+    (define-public (say-hi)
+      (ok "hello world")
+    )
 
-5. Add the address to your environment.
-
-    ```bash
-    # DEMO_ADDRESS=SP28Z69HE5H70BVRG4VGKN4SYNVJ1J0417WVCKZWM
+    (define-public (echo-number (val int))
+      (ok val)
+    )
     ```
 
-6. Launch the `tokens.clar` contract and assign it to your `DEMO_ADDRESS` address.
+    On the first line, a new public function `say-hi` is declared. To create private functions, you would use the `define-private` function. Private functions can only be executed by the current smart contract and not from the outside. Only public functions can be called from outside e.g., through other smart contracts or applications. The need to call another smart contract can have different reasons. It allows to reuse existing smart contracts published by other developers. It may also be required because complex implementations are often split out into a set of contracts to achieve a [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns).
```

Comment:
`define-private` is not a function; it's a keyword.

Reply:
`from outside e.g., through other smart contracts or applications`

might be better phrased as 

`from outside, by other smart contracts.`  Programs running outside the blockchain cannot, in general, invoke code within the blockchain (the limited exception is `define-read-only`).

Reply:
`The need to call another smart contract can have different reasons. It allows to reuse existing smart contracts published by other developers. It may also be required because complex implementations are often split out into a set of contracts to achieve a [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns).`

might be better phrased as 

`The reason public functions exist is to enable re-using code that is already available in other smart contracts, and to enable developers to break complex smart contracts into smaller, simpler smart contracts (an exercise in [separating concerns](https://en.wikipedia.org/wiki/Separation_of_concerns)).`